src/udev/udevadm-trigger.c

```diff
@@ -44,7 +50,15 @@ static int exec_list(sd_device_enumerator *e, sd_device_action_t action, Set **s
                 if (arg_dry_run)
                         continue;
 
-                r = sd_device_trigger(d, action);
+                /* Use the UUID mode if the user explicitly asked for it, or if --settle has been specified,
+                 * so that we can recognize our own uevent. */
+                r = sd_device_trigger_with_uuid(d, action, arg_uuid || settle_hashmap ? &id : NULL);
+                if (r == -EINVAL && !arg_uuid && settle_hashmap) {
+                        /* If we specified a UUID because of the settling logic, and we got EINVAL this might
+                         * be caused by an old kernel which doesn't know the UUID logic (pre-4.13). Let's try
+                         * if it works without the UUID logic then. */
+                        r = sd_device_trigger(d, action);
```

Comment:
If we enter this branch, then how about to unset `arg_uuid`? Then, we can set `NULL` for `sd_device_trigger_with_uuid()` for the subsequent devices.